nntrainer/layers/layer_node.h

```diff
@@ -41,6 +46,11 @@ class LayerNode : public ml::train::Layer, public GraphNode {
    */
   ~LayerNode() = default;
 
+  /**
+   * @brief     Set the index for the node
+   */
+  void setIndex(size_t idx) { index = idx; }
```

Comment:
Thinking about the usage of a name, I think it's better to manage name in node level as well. Maybe for the later PR?

Reply:
Yeah, I am thinking about it, as each node needs an identifier, which is both name and idx.
However, still, the name will be owned inside the layer itself, not in the LayerNode but it will be allowed to set via GraphNode directly.